include/af/array.h

```diff
@@ -682,6 +729,64 @@ namespace af
            @}
         */
 
+#ifdef AF_ENABLE_STL_CAST
+        template<typename T> array& operator=(const std::vector<T> &other)
+        {
+            //TODO: Unsafe. loses data if af_weak_copy fails
+            if(this->arr != 0) {
+                af_err err = af_release_array(this->arr);
+                if(err != AF_SUCCESS) { throw af::exception(err); }
+            }
+
+            af_array temp = 0;
+            dim_t sz = other.size();
+            af_dtype type = (af_dtype)dtype_traits<T>::af_type;
+            af_err err = af_create_array(&temp, (const void*)&other[0], 1, &sz, type);
+            if(err != AF_SUCCESS) { throw af::exception(err); }
+            this->arr = temp;
+            return *this;
+        }
+
+        template<typename T> array& operator=(std::valarray<T> &other)
+        {
+            //TODO: Unsafe. loses data if af_weak_copy fails
+            if(this->arr != 0) {
+                af_err err = af_release_array(this->arr);
+                if(err != AF_SUCCESS) { throw af::exception(err); }
+            }
+
+            af_array temp = 0;
+            dim_t sz = other.size();
+            af_dtype type = (af_dtype)dtype_traits<T>::af_type;
+            af_err err = af_create_array(&temp, &other[0], 1, &sz, type);
+            if(err != AF_SUCCESS) { throw af::exception(err); }
+            this->arr = temp;
+            return *this;
+        }
+
+        template<typename T> operator std::vector<T>() {
```

Comment:
I really really really don't want this function in arrayfire. It will allow the following code to work:

```
vector<float> my_funciton(vector<float>& foo) {
  for(auto &b : foo) {
    b++;
  }
  return foo;
}
int main () {
  af::array bar = randu(10, 10);
  bar = my_function(bar);
  return 0;
}
```